api/src/main/java/org/openmrs/util/ThreadSafeCircularFifoQueue.java

```diff
@@ -756,15 +756,13 @@ public void remove() {
 				
 				if (prevIndex == NONE) {
 					throw new IllegalStateException();
-				} else if (prevIndex >= 0) {
-					if (elements[prevIndex] == prevItem) {
+				} else if ((prevIndex >= 0) && (elements[prevIndex] == prevItem)) {
 						internalRemoveAtIndex(prevIndex);
 
-						if (prevIndex != read) {
+			      	}	else if (prevIndex != read) {
```

Comment:
Tab before tab?

Reply:
Additionally, there are spaces before the first brace?

Reply:
Finished all check again, and left some comments